x-pack/plugins/remote_clusters/public/application/sections/components/cloud_deployment_form/cloud_deployment_form.tsx

```diff
@@ -0,0 +1,186 @@
+
+  const configureCluster = () => {
+    const urlError: string | null = validateCloudUrl(cloudDeploymentUrl);
+    if (urlError) {
+      setError(urlError);
+    } else {
+      onClose();
+      onClusterConfigure(convertCloudUrlToProxyConnection(cloudDeploymentUrl));
+    }
+  };
+  return (
+    <EuiFlyout onClose={() => onClose()}>
+      <EuiFlyoutHeader hasBorder>
+        <EuiTitle>
+          <h2>
+            <EuiFlexGroup alignItems="center" gutterSize="s">
+              <EuiFlexItem grow={false}>
+                <EuiIcon type="logoCloud" size="l" />
+              </EuiFlexItem>
+
+              <EuiFlexItem>
+                <FormattedMessage
+                  id="xpack.remoteClusters.cloudDeploymentForm.formTitle"
+                  defaultMessage=" Add Cloud deployment as remote cluster"
+                />
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </h2>
+        </EuiTitle>
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>
+        <EuiText>
+          <p>
+            <FormattedMessage
+              id="xpack.remoteClusters.cloudDeploymentForm.formDescription"
+              defaultMessage="If you're connecting to a Cloud deployment, you can copy and paste the Elasticsearch
+              endpoint URL into the field below. The remote cluster will be automatically configured to connect to the Cloud deployment."
+            />
+          </p>
+        </EuiText>
+
+        <EuiSpacer />
+
+        <EuiAccordion
+          id="cloudDeploymentScreenshot"
+          buttonContent={i18n.translate(
+            'xpack.remoteClusters.cloudDeploymentForm.screenshotButtonLabel',
+            { defaultMessage: 'Expand for a screenshot' }
+          )}
+        >
+          <EuiImage url={Screenshot} alt="Screenshot of Cloud deployment url link" />
+        </EuiAccordion>
+
+        <EuiSpacer />
+
+        <EuiFormRow
+          fullWidth={true}
+          isInvalid={!!error}
+          error={error}
+          label={
+            <FormattedMessage
+              id="xpack.remoteClusters.cloudDeploymentForm.inputLabel"
+              defaultMessage="Elasticsearch endpoint URL"
+            />
+          }
+        >
+          <EuiFieldText
+            isInvalid={!!error}
+            fullWidth={true}
+            value={cloudDeploymentUrl}
+            onChange={(e) => setCloudDeploymentUrl(e.target.value)}
+          />
+        </EuiFormRow>
+
+        <EuiSpacer />
+
+        <EuiText color="subdued">
+          <p>
+            <i>
+              <strong>
+                <FormattedMessage
+                  id="xpack.remoteClusters.cloudDeploymentForm.aliasNoteLabel"
+                  defaultMessage="Note: "
+                />
+              </strong>
+              <FormattedMessage
+                id="xpack.remoteClusters.cloudDeploymentForm.aliasNoteDescription"
+                defaultMessage="If you configured a deployment alias in Elastic Cloud,
```

Comment:
"If you configured a deployment alias, either in Elastic Cloud, in your reverse proxy or load-balancer, you will need to copy-paste the **Proxy address** and **Server name** or the remote deployment in the remote cluster form directly. These values can be copied from the Remote parameters section on the remote deployment Security page. More information is available here".

As you mentioned, we can't point to the relevant security page so I think we need to only keep the link to our user guide.

@shubhaat are we going to call this feature deployment alias or configurable deployment endpoints?

Reply:
On the UI it is called an "Custom endpoint alias", we did not use the "deployment alias" terminology since at the time this was designed we still had the CCS template that used the "deployment alias" term

<img width="1407" alt="Screen Shot 2021-03-05 at 7 11 17 AM" src="https://user-images.githubusercontent.com/4782718/110499713-5cd39600-80ad-11eb-9735-64516dd023a6.png">

I suggest using "Custom endpoint alias" instead for consistency.

Reply:
Thanks @shubhaat ! 

Heads up @yuliacech @cjcenizal et al that when using the alias in the URL and changing the alias things will break. That's expected since changing the alias requires changing clients, integrations, etc. Just want to raise it in the context of a need to handle what happens when we can't connect to the cluster during and after creation.

One of the benefits of using a UUID is that it never changes.